lib/Interpreter/CppInterOp.cpp

```diff
@@ -1027,61 +1056,73 @@ namespace Cpp {
   }
 
   TCppFunction_t
-  BestTemplateFunctionMatch(const std::vector<TCppFunction_t>& candidates,
+  BestOverloadFunctionMatch(const std::vector<TCppFunction_t>& candidates,
```

Comment:
Another note is that we should check the memory growth when we call this function N times. Ideally it should be 0 for failed overload resolution candidate set. In reality it should be close to 0...